Wikipedia/Code/SearchLanguagesBarViewController.swift

```diff
@@ -57,25 +57,9 @@ class SearchLanguagesBarViewController: UIViewController, WMFPreferredLanguagesV
         otherLanguagesButton?.setTitle(WMFLocalizedString("main-menu-title", value:"More", comment:"Title for menu of secondary items.\n{{Identical|More}}"), for: .normal)
         otherLanguagesButton?.titleLabel?.font = UIFont.wmf_font(.subheadline)
         
-        NotificationCenter.default.addObserver(forName: NSNotification.Name.WMFAppLanguageDidChange, object: nil, queue: nil) { notification in
-            guard let langController = notification.object, let appLanguage = (langController as AnyObject).appLanguage else {
-                assertionFailure("Could not extract app language from WMFAppLanguageDidChangeNotification")
-                return
-            }
-            self.currentlySelectedSearchLanguage = appLanguage
-        }
+        NotificationCenter.default.addObserver(self, selector: #selector(appLanguageDidChange(_:)), name: NSNotification.Name.WMFAppLanguageDidChange, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(appLanguageDidChange(_:)), name: NSNotification.Name.WMFPreferredLanguagesDidChange, object: nil)
```

Comment:
Should this selector be `preferredLanguagesDidChange(_:)` instead of `appLanguageDidChange(_:)`

Reply:
Yes, good catch!